src/ostree/ot-builtin-commit.c

```diff
@@ -196,6 +198,13 @@ commit_filter (OstreeRepo         *self,
       g_hash_table_remove (mode_adds, path);
     }
 
+  if (opt_canonical_permissions &&
+      g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR)
```

Comment:
It could be, but we might want to commit this to an archive-z2 repo and then pull it into a bare-user-only one.